### PR TITLE
fix: add default NRC handler for unknown negative response codes

### LINKED ISSUE #60: Unknown NRC handling
Currently, when a NRC isn't defined as possible result for service, the CDA returns:
```json
{
  "message": "UDS lookup error: No matching response found for Service ID 127",
  "error_code": "vendor-specific",
  "vendor_code": "not-found"
}
``` 

If the specific NRC isn't defined, we should use a default nrc handler, that returns for all standardized NRCs a response that is compatible to our regular NRC error response.

### PR DESCRIPTION
**Pull request by ANAVHEOBA**
fix: add default NRC handler for unknown negative response codes

  Currently when an ECU returns a negative response code that isn't explicitly
  defined in the database, we just throw a generic "lookup error" which isn't
  very helpful for debugging. This change makes the CDA smarter about handling
  these cases - it now parses the raw UDS response and looks up the NRC in the
  ISO 14229 standard table to give a proper description.

  Fixes eclipse-opensovd/classic-diagnostic-adapter#60

  

  ## Summary

  Added fallback NRC handling when database mappings are missing. Instead of
  failing with a vague error, we now extract the NRC byte from the raw response
  and return a proper description using the ISO 14229 standard codes.

  For example, an unmapped NRC 0x33 will now return "Security Access Denied"
  instead of "UDS lookup error: No matching response found".

  ## Checklist

  - [x] I have tested my changes locally
  - [ ] I have added or updated documentation
  - [x] I have linked related issues or discussions
  - [x] I have added or updated tests

  ## Related

  Fixes #60

  ## Notes for Reviewers

  The main changes are in `diagservices.rs`:
  - Modified `as_nrc()` to fall back to raw data parsing when `mapped_data` is None
  - Added `extract_nrc_from_raw_data()` helper function
  - Added 7 unit tests covering various NRC values and edge cases

### CODE REVIEW

File: cda-core/src/diag_kernel/diagservices.rs [hunk 1]
```diff
@@ -372,3 +373,131 @@ impl DiagServiceResponseStruct {
...
+        let data = vec![0x7F, 0x19, 0x10];
+        let result = extract_nrc_from_raw_data(&data).unwrap();
+
+        assert_eq!(result.code, Some(0x10));
+        assert_eq!(result.sid, Some(0x19));
+        assert_eq!(result.description, Some("General Reject".to_owned()));
+    }
+
+    #[test]
+    fn test_extract_nrc_from_raw_data_response_pending() {
```
Comment (alexmohr) [inline cda-core/src/diag_kernel/diagservices.rs]: suggestion: The first three tests added here, do test pretty much the same thing with different nrcs. I would prefer only keeping one of them, as the other tests do not add any real value imho.

**Review by alexmohr — CHANGES_REQUESTED** [review]
